katcp/test/test_server.py

```diff
@@ -1014,13 +1014,26 @@ def test_sampling(self):
         self.server.ioloop.add_callback(self.server.clear_strategies, client_conn)
         self.server.sync_with_ioloop()
         self.client.assert_request_succeeds("sensor-sampling", "an.int",
-                                            args_equal=[b"an.int", b"none"])
-
+                                            args_equal=["an.int", "none"])
         # Check that we did not accidentally clobber the strategy datastructure
         # in the proccess
         self.client.assert_request_succeeds(
             "sensor-sampling", "an.int", "period", 0.125)
 
+    def test_assert_request_succeeds(self):
+        """Test exercises assert_request_succeeds"""
+        self.client.assert_request_succeeds("sensor-sampling", "an.int",
+                                            args_equal=["an.int", "none"])
+        self.client.assert_request_succeeds("sensor-sampling", b"an.int",
+                                            args_equal=["an.int", "none"])
+        self.client.assert_request_succeeds("sensor-sampling", b"an.int",
+                                            args_equal=[b"an.int", b"none"])
+        self.client.assert_request_succeeds("log-level", "debug", args_equal=["debug"],
+                                            args_echo=True)
+        self.client.assert_request_succeeds("log-level", "debug", args_equal=[b"debug"],
+                                            args_echo=True)
```

Comment:
`args_echo` takes precedence over `args_equal` - which one do we want to test on these last two asserts?